apis/config/v1beta1/configuration_types.go

```diff
@@ -349,3 +352,28 @@ type ClusterQueueVisibility struct {
 	// Defaults to 10.
 	MaxCount int32 `json:"maxCount,omitempty"`
 }
+
+// +kubebuilder:validation:Enum=LessThanOrEqualToFinalShare;LessThanInitialShare
+type PreemptionStrategy string
+
+const (
+	LessThanOrEqualToFinalShare PreemptionStrategy = "LessThanOrEqualToFinalShare"
+	LessThanInitialShare        PreemptionStrategy = "LessThanInitialShare"
+)
+
+type FairSharing struct {
+	// enabled indicates whether fair sharing is enabled for all cohorts.
+	// Defaults to false.
+	Enabled bool `json:"enabled,omitempty"`
+
+	// preemptionStrategies indicates which constraints should a preemption satisfy.
+	// The preemption algorithm will only use the next strategy in the list if the
+	// incoming workload doesn't fit after using the previous strategies.
+	// Possible values are:
+	// - LessThanOrEqualToFinalShare: Only preempt if the share of the preemptor CQ
+	//   will be less than or equal to the share of the preemptee CQ after the preemption occurs.
+	// - LessThanInitialShare: Only preempt if the share of the preemptor CQ will be strictly less
+	//   than the share of the preempte CQ before the preemption.
+	// The default strategy is ["LessThanOrEqualToFinalShare"].
+	PreemptionStrategies []PreemptionStrategy `json:"preemptionStrategies,omitempty"`
```

Comment:
I have some discussion points here. Which word should we use? Or do we not need to care if both fields are aligned with the same rule?

The first is `Strategy` vs `Policy` since we use the `Policy` for ClusterQueues. 

https://github.com/kubernetes-sigs/kueue/blob/3cbb7fcfbd27bb49964edc84c60c233b45244b30/apis/kueue/v1beta1/clusterqueue_types.go#L354

The second is `Less` vs `Lower` since we use the `Lower` for the ClusterQueue.

https://github.com/kubernetes-sigs/kueue/blob/3cbb7fcfbd27bb49964edc84c60c233b45244b30/apis/kueue/v1beta1/clusterqueue_types.go#L359-L360

Reply:
I was going for `Strategy` to match `queueingStrategy`, but `Policy` would work too. I think strategies works a little better given than the algorithm iterates through them.

Low/High works well for priority, but I don't think it works well for "share".

Reply:
That makes sense.
I agree with `strategy` and `Less/High`!